iac-extensions/docker/src/main/java/org/sonar/iac/docker/symbols/Symbol.java

```diff
@@ -45,6 +44,8 @@ public void addUsage(Scope scope, DockerTree tree, Usage.Kind kind) {
     var usage = new Usage(scope, tree, kind);
     usages.add(usage);
     if (tree instanceof HasSymbol treeWithSymbol) {
+      // Each symbol usage only references previous usages
+      // This is done to easily to compute findLastAccessibleAssignedValue in ArgumentResolution
       treeWithSymbol.setSymbol(new Symbol(this));
```

Comment:
I was confused during my implementation, so I looked why this was added to the Docker Symbol.
I found the discussion in https://github.com/SonarSource/sonar-iac/pull/586#discussion_r1129239346 and added the information as a comment.